packages/@aws-cdk/aws-fsx/lib/lustre-file-system.ts

```diff
@@ -0,0 +1,305 @@
+      });
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The default FSx file system type used by FSx for Lustre.
+   */
+  private static readonly DEFAULT_FILE_SYSTEM_TYPE: string = "LUSTRE";
+
+  /**
+   * The default port the file system listens on.
+   */
+  private static readonly DEFAULT_PORT: number = 988;
+
+  /**
+   * The security groups/rules used to allow network connections to the file system.
+   */
+  public readonly connections: Connections;
+
+  /**
+   * The ID that AWS assigns to the file system.
+   */
+  public readonly fileSystemId: string;
+
+  /**
+   * The encapsulated L1 file system.
+   */
+  private readonly fsxFileSystem: CfnFileSystem;
+
+  constructor(scope: Construct, id: string, props: FsxLustreFileSystemProps) {
+    super(scope, id);
+
+    this.validateProps(props);
+
+    const securityGroup = (props.securityGroup || new SecurityGroup(this, 'FsxLustreSecurityGroup', {
+      vpc: props.vpc
+    }));
+    securityGroup.addIngressRule(Peer.anyIpv4(), Port.tcp(FsxLustreFileSystem.DEFAULT_PORT), 'Default FSx port ingress');
+    securityGroup.addEgressRule(Peer.anyIpv4(), Port.tcp(FsxLustreFileSystem.DEFAULT_PORT), 'Default FSx port egress');
+
+    this.connections = new Connections({
+      securityGroups: Array.of(securityGroup),
+      defaultPort: Port.tcp(FsxLustreFileSystem.DEFAULT_PORT)
+    });
+
+    this.fsxFileSystem = new CfnFileSystem(this, 'FsxLustreFileSystem', {
+      fileSystemType: FsxLustreFileSystem.DEFAULT_FILE_SYSTEM_TYPE,
+      subnetIds: Array.of(props.vpcSubnet.subnetId),
+      backupId: props.backupId,
+      kmsKeyId: (props.kmsKey ? props.kmsKey.keyId : undefined),
+      lustreConfiguration: props.lustreConfiguration,
+      securityGroupIds: Array.of(securityGroup.securityGroupId),
+      storageCapacity: props.storageCapacity
+    });
+
+    this.fileSystemId = this.fsxFileSystem.ref;
+    this.node.defaultChild = this.fsxFileSystem;
+  }
+
+  /**
+   * Validates the props provided for a new FSx for Lustre file system.
+   */
+  private validateProps(props: FsxLustreFileSystemProps) {
+    const lustreConfiguration = props.lustreConfiguration;
+    const deploymentType = lustreConfiguration.deploymentType;
+
+    this.validateExportPath(lustreConfiguration.exportPath, lustreConfiguration.importPath);
+    this.validateImportedFileChunkSize(lustreConfiguration.importedFileChunkSize);
+    this.validateImportPath(lustreConfiguration.importPath);
+    this.validatePerUnitStorageThroughput(deploymentType, lustreConfiguration.perUnitStorageThroughput);
+    this.validateStorageCapacity(deploymentType, props.storageCapacity);
+    this.validateWeeklyMaintenanceStartTime(lustreConfiguration.weeklyMaintenanceStartTime);
+  }
+
+  /**
+   * Validates the export path is in the correct format and matches the import path.
+   */
+  private validateExportPath(exportPath?: string, importPath?: string): void {
+    if (exportPath === undefined) { return; }
```

Comment:
If either value is `Token.isUnresolved()` then you can't really validate, in which case you need to early return as well. This holds for all validation routines.

Reply:
I've implemented this check for import path and export path. After researching Tokens, those were the two fields that seemed likely to contain unresolved Tokens, since they can both be S3 buckets.

It was a little tricky trying to work through all the valid use cases for export path. I decided it made the most sense that if export path is set as an unresolved Token, import path would also need to be, but I'm really unsure about that.